Bugfix: Change zk-link-regexp to macro of the same name

boyechko opened the pull request:
This way zk-link-regexp is generated at run-time, so it always reflects user
changes to zk-link-format and zk-id-regexp.

localauthor commented on zk.el:
```diff
@@ -531,25 +535,22 @@ Adds `zk-make-link-buttons' to `find-file-hook.'"
-          (let ((beg (match-beginning 1))
-                (end (match-end 1))
-                (id (match-string-no-properties 1)))
-            (when (member id ids)
```
Why remove this check?

localauthor commented on zk.el:
```diff
@@ -720,21 +721,16 @@ Optionally call a custom function by setting the variable
-          ((eq 1 (length id-list))
-           (list (zk--parse-id 'file-path id-list)))
-          (t
-           (zk--parse-id 'file-path (delete-dups id-list))))))
```
Not clear how these changes are related to the bugfix

localauthor commented:
This is a good idea, for sure. I'd like to merge it, but I'm not sure why the id checks are removed, as commented above.

boyechko commented:
I did explain my reasoning about the checks in the review comments, but those changes are unrelated to zk-link-regexp, so don't really belong in this pull request anyway.

I reverted the changes and, more importantly, fixed the macro definition causing obscure issues it was meant to solve. It's been a while since I wrote lisp macros; sorry. It's correct now.

localauthor commented:
> I did explain my reasoning about the checks in the review comments, but those changes are unrelated to zk-link-regexp, so don't really belong in this pull request anyway.

Strange I don't see any responses to my review comments. Maybe a github issue?

> I reverted the changes and, more importantly, fixed the macro definition causing obscure issues it was meant to solve. It's been a while since I wrote lisp macros; sorry. It's correct now.
